src/main/scala/nl.knaw.dans.easy.validatebag/EasyValidateDansBagApp.scala

```diff
@@ -57,13 +58,41 @@ class EasyValidateDansBagApp(configuration: Configuration) extends DebugEnhanced
   }
 
   def validate(uri: URI, infoPackageType: InfoPackageType): Try[ResultMessage] = {
-    for {
+    val bagName = resolveAndLogBagName(uri)
+    val resultMessage = for {
       bag <- getBagPath(uri)
       version <- getProfileVersion(bag)
       violations <- validation.checkRules(new TargetBag(bag, version), allRules(version), infoPackageType)(isReadable = _.isReadable)
         .map(_ => Seq.empty)
         .recoverWith(extractViolations)
     } yield ResultMessage(uri, bag.getFileName.toString, version, infoPackageType, violations)
+    logResult(bagName, resultMessage)
```

Comment:
since you're only using the `violations` from the whole `resultMessage`, why don't you integrate this call in the for-comprehension. Your `logResult` function becomes much simpler that way.

```scala
for {
...
violations <- ...
_ = logResult(bagName, violations)
} yield ResultMessage(...)
```